envs/monkey_zoo/blackbox/island_client/island_config_parser.py

```diff
@@ -1,29 +1,17 @@
-import json
+from typing import Iterable, Mapping
 
 import dpath.util
-from typing_extensions import Type
 
-from envs.monkey_zoo.blackbox.config_templates.config_template import ConfigTemplate
-from envs.monkey_zoo.blackbox.island_client.monkey_island_client import MonkeyIslandClient
+from common.configuration.agent_configuration import AgentConfiguration
 
 
 class IslandConfigParser:
     @staticmethod
-    def get_raw_config(
-        config_template: Type[ConfigTemplate], island_client: MonkeyIslandClient
-    ) -> str:
-        response = island_client.get_config()
-        config = IslandConfigParser.apply_template_to_config(
-            config_template, response["configuration"]
-        )
-        return json.dumps(config)
-
-    @staticmethod
-    def apply_template_to_config(config_template: Type[ConfigTemplate], config: dict) -> dict:
-        for path, value in config_template.config_values.items():
-            dpath.util.set(config, path, value, ".")
-        return config
+    def get_serialized_config(agent_configuration: AgentConfiguration) -> str:
+        return agent_configuration.to_json()
```

Comment:
I don't think this method is necessary. We can modify test_blackbox.py to have `serialized_config = agent_configuration.to_json()`.